compiler/rustc_typeck/src/collect.rs

```diff
@@ -2747,7 +2747,16 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, id: DefId) -> CodegenFnAttrs {
         } else if attr.has_name(sym::rustc_std_internal_symbol) {
             codegen_fn_attrs.flags |= CodegenFnAttrFlags::RUSTC_STD_INTERNAL_SYMBOL;
         } else if attr.has_name(sym::used) {
-            codegen_fn_attrs.flags |= CodegenFnAttrFlags::USED;
+            let inner = attr.meta_item_list();
+            match inner.as_deref() {
+                Some([item]) if item.has_name(sym::retain) => {
```

Comment:
For the gating, I think you'd just add a `if !tcx.features().used_with_arg { feature_err(...).emit(); }` in here, so it only applies to the case where an argument is passed.

Reply:
Ahh, I see. Thanks!